app/models/manageiq/providers/hawkular/middleware_manager/alert_manager.rb

```diff
@@ -0,0 +1,85 @@
+module ManageIQ::Providers
+  class Hawkular::MiddlewareManager::AlertManager
+    require 'hawkular/hawkular_client'
+
+    def initialize(ems)
+      @alerts_client = ems.connect.alerts
+    end
+
+    def process_alert(operation, miq_alert)
+      group_trigger = convert_group_trigger(miq_alert)
+      group_conditions = convert_group_conditions(miq_alert)
+      case operation
+      when :new
+        @alerts_client.create_group_trigger(group_trigger)
+        @alerts_client.set_group_conditions(group_trigger.id,
+                                            :FIRING,
+                                            group_conditions)
+      when :update
+        @alerts_client.update_group_trigger(group_trigger)
+        @alerts_client.set_group_conditions(group_trigger.id,
+                                            :FIRING,
+                                            group_conditions)
+      when :delete
+        @alerts_client.delete_group_trigger(group_trigger.id)
+      end
+    end
+
+    def convert_group_trigger(miq_alert)
```

Comment:
I read it confusing from that side.
process_alert() converts first a miq_alert into a group_trigger that is later created using the @alerts_client.create_group_trigger().

Convert is synonym of transform/parse.